wordpress-mobile/WordPress-MediaPicker-Android
# [Build] Upgrade Gradle to 8.2.1 and AGP to 8.1.0

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

This PR upgraded Gradle to [8.2.1](https://docs.gradle.org/8.2.1/release-notes.html) and AGP to [8.0.2](https://developer.android.com/build/releases/gradle-plugin#agp-8-0-2).

FYI: The [8.1.1](https://docs.gradle.org/8.1.1/release-notes.html) Gradle upgrades had to happen in 2 parts (bfb8e53523f3c2e2a915774af1353f503b11e415 and 82780f130d60b8a0dba35d383ba06532462af0d9). This is because of the breaking changes that this upgrade introduces, including the `AGP` upgrade and its associated breaking changes (namespace, build option, etc).

-----

🚫 Blocked By:


<details>
    <summary>CI Failure</summary>

```
[2023-07-07T14:01:28Z] FAILURE: Build failed with an exception.
--
  | [2023-07-07T14:01:28Z]
  | [2023-07-07T14:01:28Z] * Where:
  | [2023-07-07T14:01:28Z] Build file '/var/lib/buildkite-agent/builds/ci-android-i-0c8ad4134fec7403e-1/automattic/wordpress-mediapicker-android/mediapicker/build.gradle' line: 2
  | [2023-07-07T14:01:28Z]
  | [2023-07-07T14:01:28Z] * What went wrong:
  | [2023-07-07T14:01:28Z] An exception occurred applying plugin request [id: 'com.android.library', version: '8.0.2']
  | [2023-07-07T14:01:28Z] > Failed to apply plugin 'com.android.internal.library'.
  | [2023-07-07T14:01:28Z]    > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
  | [2023-07-07T14:01:28Z]       Your current JDK is located in /usr/lib/jvm/java-11-openjdk-11.0.19.0.7-1.amzn2.0.1.x86_64
  | [2023-07-07T14:01:28Z]       You can try some of the following options:
  | [2023-07-07T14:01:28Z]        - changing the IDE settings.
  | [2023-07-07T14:01:28Z]        - changing the JAVA_HOME environment variable.
  | [2023-07-07T14:01:28Z]        - changing `org.gradle.java.home` in `gradle.properties`.
```

</details>

-----

Breaking Change List:

1. [Move package from android manifest to build files](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/a8824d098721d1d00da58e3c8a7386a74ca587e4)
2. [Enable build config build feature for sampleapp module](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/4c2f71611644b066fb7e22b21f14fea46d6b7082)

AGP Upgrade Assistant List:

1. [Remove ignored configuration properties](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/f56e78357486a3fccfa9bd5b5715c7ef1fe753e2)
2. [Migrate dsl properties from lint options to lint](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/27818ae7746ccb0ab5bb125b24e873550ce0425d)

Lint Resolution List:

1. [Resolve notifications without permission lint warning](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/18c353a1e59d5eab5cf749ae8ab878bb3cbfba05)

-----

## Associated clients

You can reference the `[Build] Upgrade Gradle to 8.1.1 and AGP to 8.0.2` PRs of the below associated clients for testing/verification purposes:

- [ ] WCAndroid#<PR_NUMBER>

-----

## To test

1. Verify that all the CI checks are successful.
2. Smoke test the `sampleapp` example app.
3. Test composite builds via the associated client apps (see above).

-----

## Merge instructions
